fim/slivers/component_catalog.py

```diff
@@ -109,10 +109,12 @@ def generate_component(self, *, name: str, ctype: ComponentType, model: str,
                     isliver.node_id = str(uuid.uuid4())
                 if interface_labels is not None:
                     isliver.set_labels(interface_labels[id_index])
+                # if labels are lists, extract the length to make it the number of units
```

Comment:
This will set the number of units > 1 even for the Smart NICs. Is this intentional?

Reply:
So it actually doesn't because by the time the labels get to the interface, the list is split on per interface basis, so unit will be one: 
```
{
  "identity": 597,
  "labels": [
    "ConnectionPoint",
    "GraphNode"
  ],
  "properties": {
"LabelDelegations": "{"primary": {"pool_id": "_", "labels": {"mac": "04:3F:72:B7:19:5D", "vlan_range": "1-4096"}}}",
"Type": "TrunkPort",
"CapacityDelegations": "{"primary": {"pool_id": "_", "capacities": {"bw": 100, "unit": 1}}}",
"GraphID": "d8a28da3-b175-410c-a47e-30720090f09d",
"NodeID": "node_id-04-3F-72-B7-19-5D",
"Class": "ConnectionPoint",
"Labels": "{"mac": "04:3F:72:B7:19:5D", "vlan_range": "1-4096"}",
"id": "522",
"Capacities": "{"bw": 100, "unit": 1}",
"StitchNode": "false",
"Name": "renc-w2-nic2-p2"
  }
}
```

Reply:
got it